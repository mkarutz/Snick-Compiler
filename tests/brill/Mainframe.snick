proc main ()
    int num;

    write "Enter the number of disks: ";
    read num;
    write "The sequence of moves involved in the Tower of Hanoi are:\n";
    towers(num, 1, 3, 2);
end

proc towers (val int num, val int frompeg, val int topeg, val int auxpeg)
    if num != 1 then
        towers(num - 1, frompeg, auxpeg, topeg);
        write "Move disk from peg ";
        write frompeg;
        write " to peg ";
        write topeg;
        write "\n";
        towers(num - 1, auxpeg, topeg, frompeg);
    else
        write "Move disk from peg ";
        write frompeg;
        write " to peg ";
        write topeg;
        write "\n";
    fi
end
