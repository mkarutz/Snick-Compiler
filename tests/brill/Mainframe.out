Enter the number of disks: The sequence of moves involved in the Tower of Hanoi are:
Move disk from peg 1 to peg 3
Move disk from peg 1 to peg 2
Move disk from peg 3 to peg 2
Move disk from peg 1 to peg 3
Move disk from peg 2 to peg 1
Move disk from peg 2 to peg 3
Move disk from peg 1 to peg 3
